addons/l10n_de/migrations/2.0/pre-migrate.py

```diff
@@ -25,3 +25,12 @@ def migrate(cr, version):
     rename_tag(cr, "tag_de_liabilities_bs_D_8", "tag_de_liabilities_bs_C_8")
     rename_tag(cr, "tag_de_liabilities_bs_E", "tag_de_liabilities_bs_D")
     rename_tag(cr, "tag_de_liabilities_bs_F", "tag_de_liabilities_bs_E")
+
+    # By deleting tag B and F from ir_model_data we ensure that the ORM won't try to remove those records.
+    # This is done because the tags might be already used as a FK somewhere else.
+    cr.execute(
+        """DELETE FROM ir_model_data
+           WHERE module='l10n_de'
+           AND name IN ('tag_de_liabilities_bs_B', 'tag_de_liabilities_bs_F')
+        """
+    )
```

Comment:
@aj-fuentes Given the reasoning from the description why not use `util.delete_unused`?

```suggestion
    cr.execute(
        """DELETE FROM ir_model_data
           WHERE module='l10n_de'
           AND name IN ('tag_de_liabilities_bs_B', 'tag_de_liabilities_bs_F')
        """
    )
```

Reply:
util is not available to use here as this is not in the upgrade repo